src/librustc_incremental/calculate_svh/svh_visitor.rs

```diff
@@ -25,25 +25,33 @@ use rustc::hir::def::{Def, PathResolution};
 use rustc::hir::def_id::DefId;
 use rustc::hir::intravisit as visit;
 use rustc::hir::intravisit::{Visitor, FnKind};
-use rustc::hir::map::DefPath;
 use rustc::ty::TyCtxt;
+use rustc::util::nodemap::DefIdMap;
 
 use std::hash::{Hash, SipHasher};
 
 pub struct StrictVersionHashVisitor<'a, 'tcx: 'a> {
     pub tcx: TyCtxt<'a, 'tcx, 'tcx>,
     pub st: &'a mut SipHasher,
+
+    // collect a deterministic hash of def-ids that we have seen
+    def_id_hashes: DefIdMap<u64>,
```

Comment:
I'd call this `def_path_hashes_cache` to make it clear, that this is a performance optimization.
It would make sense to share this map between invocations of the visitor (i.e. to allocate it once when the ICH-map is built and always give the visitors a reference to the same instance). That would reduce the number of memory allocations and presumably increase the number of cache hits.

More importantly, I'd document that this contains the hashes of just the DepPaths, not of the items the DefIds designate.